lib/docs/filters/bluebird/entries.rb

```diff
@@ -0,0 +1,66 @@
+module Docs
+  class Bluebird
+    class EntriesFilter < Docs::EntriesFilter
+      TYPE_MAP = {
+        Core: %w(new-promise then spread catch error finally bind promise.join
+          promise.try promise.method promise.resolve promise.reject core
+          promise.bind),
+        'Synchronous inspection': %w(promiseinspection isfulfilled isrejected
+          ispending iscancelled value reason),
+        Collections: %w(promise.all promise.props promise.any promise.some
+          promise.map promise.reduce promise.filter promise.each
+          promise.mapseries promise.race all props any some map reduce filter
+          each mapseries),
+        'Resource management': %w(promise.using disposer),
+        Promisification: %w(promise.promisify promise.promisifyall
+          promise.fromcallback ascallback),
+        Timers: %w(delay timeout promise.delay),
+        Cancellation: %w(cancel),
+        Generators: %w(promise.coroutine promise.coroutine.addyieldhandler),
+        Utility: %w(tap tapcatch call get return throw catchreturn catchthrow
+          reflect promise.getnewlibrarycopy promise.noconflict
+          promise.setscheduler),
+        'Built-in error types': %w(operationalerror timeouterror
+          cancellationerror aggregateerror),
+        Configuration: %w(global-rejection-events local-rejection-events
+          done promise.config suppressunhandledrejections
+          promise.onpossiblyunhandledrejection promise.bind
+          promise.onunhandledrejectionhandled),
+        'Progression migration': %(),
+        'Deferred migration': %(),
+        'Environment variables': %(),
+        "Beginner's Guide": %w(),
+        'Error management configuration': %w(),
+        'Anti-patterns': %w(),
+        'Deprecated APIs': %w()
+      }
+
+      def get_name
+        name = at_css('h1.post-title')
+        if name.nil?
+          name = at_css('.post-content h2')
+        end
+        name.text
+      end
+
+      def get_type
+        type = nil
+        TYPE_MAP.each do |k,v|
+          if k.to_s.casecmp(name.strip) == 0
+            type = k
+            break
+          else
+            slug_end = slug.sub(%r(^docs/api/), '')
+            if v.include?(slug_end.downcase)
+              type = k
+              break
+            end
+          end
+        end
+
+        type.to_s
```

Comment:
For a non-rubyist, what exactly is going on with the keys of the `TYPE_MAP` hash? Both here and [up above on line 49](https://github.com/Thibaut/devdocs/pull/684/files#diff-5b23a8771aed8bfa224b7f36bca446f7R49) the `to_s` was required to get things working the way I would naively expect:

* `if k.casecmp(name.strip)` would NEVER equal 0, even when `puts` was telling me that `name.strip` was a key that was _right there_ in the `TYPE_MAP` hash, like "Core"

* returning just `type` was showing up in the `thor docs:generate` as a single type instead of the 20-ish I would expect (one type per key in `TYPE_MAP`)

Reply:
The keys of the `TYPE_MAP` hash are [`Symbol`](http://devdocs.io/ruby~2.4/symbol)s, which are similar to strings but not the same. If you want the hash to have string keys, something like this should work:

```ruby
TYPE_MAP = {
  'Core' => %w(new-promise then spread ...),
  'Synchronous inspection' => %w(promiseinspection isfulfilled isrejected ...)
}
```

Reply:
Oh, the operator _after_ the key determines whether the key is a symbol or string. Thanks @j-f1: I was able to figure out the fact that the keys were ending up symbols instead of strings, but could not figure out why.

Is it more accepted to separate keys and values in a hash declaration fat arrows rather than colons in Ruby? I see code elsewhere doing `{ :key => "value" }`, which makes it immediately clear what is going on. I can also imagine that `{ key: "value" }` is a nicer way of writing the exact same thing, though.

Reply:
> Is it more accepted to separate keys and values in a hash declaration fat arrows rather than colons in Ruby?

It’s the best option if you want the keys to be symbols. If you want them to be strings, you have to use the hash rocket (`=>`) syntax.

Reply:
> hash rocket 

*There* is the term I want to google. Thanks again, @j-f1.